packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -620,4 +608,18 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
 
     return publicKeys[0]
   }
+
+  async verifySignTx(tx: SignTx<T>) {
+    const addresses = (
+      await Promise.all(
+        tx.inputs.map(async input => {
+          const tx = await this.providers.http.getTransaction(input)
```

Comment:
We should be able to introspect inputs and outputs from the build UTXOs themselves, no need to fetch anything here

Reply:
perhaps we can pile on this in your am to discuss?